datafusion/datasource-parquet/src/opener.rs

```diff
@@ -85,7 +86,49 @@ pub(super) struct ParquetOpener {
 }
 
 impl FileOpener for ParquetOpener {
-    fn open(&self, file_meta: FileMeta) -> Result<FileOpenFuture> {
+    fn open(&self, file_meta: FileMeta, file: PartitionedFile) -> Result<FileOpenFuture> {
+        let predicate_creation_errors = MetricBuilder::new(&self.metrics)
+            .global_counter("num_predicate_creation_errors");
+        // Prune using known statistics
+        match (&file.statistics, &self.predicate) {
+            (Some(stats), Some(predicate)) => {
```

Comment:
Given that there is only one branch, I suggest using `if let (Some(_), Some(_)) = xxx {}` here.